src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -66,27 +66,83 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
         {
             if (Options.RemoteSignOutPath.HasValue && Options.RemoteSignOutPath == Request.Path)
             {
-                var remoteSignOutContext = new RemoteSignOutContext(Context, Options);
-                await Options.Events.RemoteSignOut(remoteSignOutContext);
+                return await HandleRemoteSignOutAsync();
+            }
+            return await base.HandleRequestAsync();
+        }
+
+        protected virtual async Task<bool> HandleRemoteSignOutAsync()
+        {
+            OpenIdConnectMessage message = null;
+
+            if (string.Equals(Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
+            {
+                message = new OpenIdConnectMessage(Request.Query.Select(pair => new KeyValuePair<string, string[]>(pair.Key, pair.Value)));
+            }
+            // assumption: if the ContentType is "application/x-www-form-urlencoded" it should be safe to read as it is small.
+            else if (string.Equals(Request.Method, "POST", StringComparison.OrdinalIgnoreCase)
+              && !string.IsNullOrEmpty(Request.ContentType)
+              // May have media/type; charset=utf-8, allow partial match.
+              && Request.ContentType.StartsWith("application/x-www-form-urlencoded", StringComparison.OrdinalIgnoreCase)
+              && Request.Body.CanRead) {
+                var form = await Request.ReadFormAsync();
+                message = new OpenIdConnectMessage(form.Select(pair => new KeyValuePair<string, string[]>(pair.Key, pair.Value)));
```

Comment:
Wrapping the query or form in an OIDC message just gets you named properties for the keys, but what keys are you expecting? Only `sid` right? Which isn't even on the OIDCMessage. Are you expecting any other OIDC fields? If not this seems like the wrong abstraction for this endpoint.